crypto/pkcs11key/key.go

```diff
@@ -30,6 +30,24 @@ var hashPrefixes = map[crypto.Hash][]byte{
 	crypto.RIPEMD160: {0x30, 0x20, 0x30, 0x08, 0x06, 0x06, 0x28, 0xcf, 0x06, 0x03, 0x00, 0x31, 0x04, 0x14},
 }
 
+// Ctx defines the subset of pkcs11.Ctx's methods that we use, so we can inject
+// a different Ctx for testing.
+type Ctx interface {
```

Comment:
I think we're on the same page, but to be clear: We're not talking about having pkcs11key implement an `init()` function, but about making it the caller's responsibility to import `miekg/pkcs11` and call `pkcs11.New()` and `ctx.Initialize()`, then pass the result to `pkcs11key`. The alternative, the way it currently works, is to leave responsibility for loading the module DLL in the pkcs11key library.